src/scripts/main.js

```diff
@@ -1,3 +1,33 @@
 'use strict';
 
-// write code here
+const listElement = document.querySelector('ul');
+const workersArr = [...listElement.children];
+
+function parseSalary(salaryString) {
+  return +salaryString.replace(/[$,]/g, '');
+}
+
+function sortList(arr) {
+  const sortedList = arr.sort((a, b) => {
+    const salary1 = parseSalary(a.dataset.salary);
+    const salary2 = parseSalary(b.dataset.salary);
+
+    return salary2 - salary1;
+  });
+
+  listElement.append(...sortedList);
+}
+
+function getEmployees(arr) {
+  return arr.map((item) => {
+    return {
+      name: item.dataset.innerText,
```

Comment:
The property `dataset.innerText` is likely incorrect. If you want to access the `name` attribute from the dataset, it should be `dataset.name` instead of `dataset.innerText`. Ensure that the HTML elements have a `data-name` attribute.